lib/pbench/server/database/database.py

```diff
@@ -27,12 +27,18 @@ def init_db(server_config, logger):
         if not hasattr(Database.Base, "logger"):
             Database.Base.logger = logger
 
+        # WARNING:
         # Make sure all the models are imported before this function gets called
         # so that they will be registered properly on the metadata. Otherwise
         # metadata will not have any tables and create_all functionality will do nothing
 
-        Database.Base.query = Database.db_session.query_property()
-        Database.db_session.remove()  # Remove the placeholder static session
+        # We create a static session placeholder with the Class, but we need
+        # to bind the session to our engine. Creating the class attribute here
+        # would not be viewed kindly by Python; to get this working, remove the
```

Comment:
I don't understand the comment about Python's views -- AFAIK, if we can reference the class, we can modify its attributes, here or anywhere else.  (Perhaps you mean Black or PyFlakes or something wouldn't like it?)

That said, this comment doesn't mention _why_ we create a static session placeholder.  It seems weird that we do so only to remove it, again.  If all we needed it for was to make the `query_property()` call, perhaps we don't need to do it at all, now?  (Skipping that would save having to remove it, again....)

Reply:
You know, at one point I tried to do that, removing the static session and trying to dynamically add it. Something went wrong; I gave up, reverted, and moved on.

So now I had to go back and try it again to reconstruct exactly what went wrong, and I'm finding nothing. At least, the unit tests all run correctly.

I'm going to build a new RPM and deploy it just to make sure this works correctly for real; it probably will, and if so I'll push another commit.

Reply:
And that worked, too, so I have no idea what problem I ran into before and I probably just messed it up. Fixed now.